wire.md

```diff
@@ -0,0 +1,1008 @@
+
+This request can be used to terminated previously sent, long-lived requests.
+
+field        | type                | desc
+-------------|---------------------|-------------------------------------
+`cancel_id`  | `u8[4]`             | the `req_id` of the request to be cancelled
+
+Receiving this request indicates that any further responses sent back with a
+`req_id` matching the given `cancel_id` will be ignored and discarded.
+
+`msg_type` MUST be set to `3`.
+
+`cancel_id` MUST be set to the `req_id` of the request to be cancelled.
+
+There MUST NOT be any responses to this request.
+
+Like any other request, this request MUST have its own unique `req_id` in order
+to function as intended. `cancel_id` is used to set the request identifier to
+cancel, not the `req_id`.
+
+A peer receiving a Cancel Request MUST forward it along the same route and
+peers it forwarded the original message with `req_id = cancel_id`, to the same
+peers as the original request, so that all peers who know of the original
+request are notified. This request's `ttl` MUST be ignored, in order to ensure
+all original recipients of the original request are reached.
+
+##### 6.3.2.4 Channel Time Range Request
+
+Request chat messages and chat message deletions written to a channel between a
+start and end time, optionally subscribing to future chat messages.
+
+field          | type               | desc
+---------------|--------------------|----------------------------
+`channel_len`  | `varint`           | length of the channel's name, in bytes
+`channel`      | `u8[channel_len] ` | channel name (UTF-8)
+`time_start`   | `varint`           | milliseconds since UNIX Epoch (inclusive)
+`time_end`     | `varint`           | milliseconds since UNIX Epoch (exclusive)
+`limit`        | `varint`           | maximum number of hashes to return
+
+`msg_type` MUST be set to `4`.
+
+A responder receiving this request MUST respond with 1 or more Hash Response
+messages.
+
+`time_start` is the post with the oldest timestamp the requester is interested
+in; `time_end` is the newest.
+
+A responder SHOULD include the hashes of all known `post/text` and
+`post/delete` posts made to a channel between `time_start` and `time_end`.
+
+A `time_end` of 0 MUST be understood as a request to keep this request alive
+even after all known hashes in the range `time_start` to `now()` are provided,
+and continue to receive any new chat messages that the responder learns of in
+the future, so long as this request is still alive. `time_start` MUST be
+respected regardless of whether `time_end` is 0 or some definite value.
+
+Responding clients SHOULD respond with all known chat messages within the
+requested time range, though they may desire not to in certain circumstances,
+particularly if a channel has a very long history and the responding client
+lacks sufficient resources at the time to return thousands or hundreds of
+thousands of chat message hashes.
```

Comment:
Any recommendation on which messages to transmit then (probably the newest ones?)?

Reply:
I appreciate this catch. Specifying this makes pagination possible!

>  responder is RECOMMENDED to send posts in reverse chronological order by post timestamp, so that, if the requester sees that they received a number of hashes equal to their `limit`, they can be assured that they now have all posts newer than the oldest post hash the responder did send, and can make subsequent requests to paginate backwards in time.